File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    q = deque(prices)
    
    answer = []
    while q:
        item = q.popleft()
        
        cnt = 0
        for i in q:
            cnt += 1
            if item > i:
                break
        answer.append(cnt)

    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] prices) {
        Queue<Integer> q = new LinkedList<>();
            
        for (int p: prices) {
            q.offer(p);
        }
                
        List<Integer> answer = new ArrayList<>();
        while (!q.isEmpty()) {
            int item = q.poll();
            
            int cnt = 0;
            for (int next: q) {
                cnt++;
                if (item <= next) {
                    continue;
                } else {
                    break;
                }
            }
            answer.add(cnt);
        }
        return answer;
    }
}